ui/prometheus-plugin/src/components/PromQL.tsx

```diff
@@ -14,22 +14,62 @@
 import CodeMirror, { ReactCodeMirrorProps } from '@uiw/react-codemirror';
 import { PromQLExtension, CompleteConfiguration } from '@prometheus-io/codemirror-promql';
 import { EditorView } from '@codemirror/view';
-import { useTheme, InputLabel, Stack } from '@mui/material';
-import { useMemo } from 'react';
+import { useTheme, CircularProgress, InputLabel, Stack, IconButton, Tooltip } from '@mui/material';
+import FileTreeIcon from 'mdi-material-ui/FileTree';
+import { useMemo, useState } from 'react';
+import { ErrorAlert } from '@perses-dev/components';
+import CloseIcon from 'mdi-material-ui/Close';
+import { useReplaceVariablesInString } from '@perses-dev/plugin-system';
+import { PrometheusDatasourceSelector } from '../model';
+import { useParseQuery } from './parse';
+import TreeNode from './TreeNode';
 
-export type PromQLEditorProps = { completeConfig: CompleteConfiguration } & Omit<
-  ReactCodeMirrorProps,
-  'theme' | 'extensions'
->;
+export type PromQLEditorProps = {
+  completeConfig: CompleteConfiguration;
+  datasource: PrometheusDatasourceSelector;
+} & Omit<ReactCodeMirrorProps, 'theme' | 'extensions'>;
 
-export function PromQLEditor({ completeConfig, ...rest }: PromQLEditorProps) {
+export function PromQLEditor({ completeConfig, datasource, ...rest }: PromQLEditorProps) {
   const theme = useTheme();
   const isDarkMode = theme.palette.mode === 'dark';
+  const [isTreeViewVisible, setTreeViewVisible] = useState(false);
 
   const promQLExtension = useMemo(() => {
     return new PromQLExtension().activateLinter(false).setComplete(completeConfig).asExtension();
   }, [completeConfig]);
 
+  const queryExpr = useReplaceVariablesInString(rest.value);
+
+  // Specific errors that user should be able to hide
```

Comment:
I'm not fan of that part, it seems you automatically hide the tree view in case of some errors happen, checking the error message but error messages is not something on which we could rely. Possibly did you checked if you couldn't rely on something more fixed like status code or something? Also I believe that some proxy could send different things. Let's say prometheus is behind a proxy that want to secure changing the 404 to a 403, with a message different than this one, then you're screwed.

Reply:
What if we just keep the button, tree view not visible by default. And if it is clicked, wrap any error message like that:

```
Looks like the tree view failed with following message:
---
<any error message as it comes>
---
if it is a Not Found or Permission Denied message, it might be because you're not targetting a Prometheus 3.0 datasource.
```

Reply:
Yeah I know this is not ideal and actually I didn' t manage to base the behavior on error status code instead, I dont have access to such attribute it seems...

Reply:
Tackled offline, agreed to let it like that for now & simplify the logic the day we have a real PR dedicated to https://github.com/perses/perses/issues/1419. Added comment in the code to add this context.